### ISSUE TITLE
Validate index definitions

### ISSUE THREAD

**Issue by bartv** [inmanta/inmanta#212]
Indexes can be defined on unexisting attributes and relations. An error is only produced when such an entity is constructed.

```
  File "/home/bart/.virtualenvs/inmanta/lib/python3.5/site-packages/inmanta/ast/statements/generator.py", line 146, in normalize
    raise TypingException(self, "%s is part of an index and should be set in the constructor." % attr)
inmanta.ast.TypingException: sdfsdfs is part of an index and should be set in the constructor. 
```

**Comment by bartv** [comment]
Now it is impossible to define an index over relations. Only attributes are now allowed.

**Comment by wouterdb** [comment]
should be fixed in 1ce27b868f81fcdd248831c10b4c19ec838c8e08
